Penteract/Assets/Scripts/UltimateFang.cpp

```diff
@@ -0,0 +1,87 @@
+#include "UltimateFang.h"
+
+#include "GameObject.h"
+#include "GameplaySystems.h"
+#include "Modules/ModulePhysics.h"
+#include "Components/ComponentBoundingBox.h"
+
+EXPOSE_MEMBERS(UltimateFang) {
+	MEMBER(MemberType::FLOAT, radius),
+	MEMBER(MemberType::PREFAB_RESOURCE_UID, fangBulletUID),
+	MEMBER(MemberType::PREFAB_RESOURCE_UID, fangTrailUID)
+};
+
+GENERATE_BODY_IMPL(UltimateFang);
+
+void UltimateFang::Start() {
+
+	trail = GameplaySystems::GetResource<ResourcePrefab>(fangTrailUID);
+	bullet = GameplaySystems::GetResource<ResourcePrefab>(fangBulletUID);
+
+	GetOwner().Disable();
+}
+
+void UltimateFang::Update() {
+	tickCurrent -= Time::GetDeltaTime();
+	if (tickCurrent <= 0) {
+		tickOn = true;
+		tickCurrent = tickDuration;
+	}
+	else {
+		tickOn = false;
+		if (collisionedGameObject.size() > 0) {
+			int i = rand() % collisionedGameObject.size(); // select random enemi from the list
+			ComponentTransform* transformOwner = GetOwner().GetComponent<ComponentTransform>();
+			ComponentTransform* transformTarget = collisionedGameObject[i].GetComponent<ComponentTransform>();
+
+			GameObject* fang = GetOwner().GetParent();
+
+			ComponentCapsuleCollider* ccOwner = fang->GetComponent<ComponentCapsuleCollider>();
+			float midHeight = 0.0f;
+			if(ccOwner){
+				midHeight = ccOwner->height;//bbOwner->GetLocalMaxPointAABB().y / 2;
+			}
+
+			float3 posFang = transformOwner->GetGlobalPosition();
+			float3 posTarget = transformTarget->GetGlobalPosition();
+			float3 dir = float3(posTarget.x - posFang.x, posTarget.y - posFang.y, posTarget.z - posFang.z).Normalized();
+			GameplaySystems::Instantiate(bullet, transformOwner->GetGlobalPosition() + float3(0, midHeight, 0), DirectionToQuat(dir));
+			GameplaySystems::Instantiate(trail,  transformOwner->GetGlobalPosition() + float3(0, midHeight, 0), DirectionToQuat(dir));
+		}
+	}
+	collisionedGameObject.clear();
+}
+
+Quat UltimateFang::DirectionToQuat(float3 dir) {
+	float angle = atan2(dir.x, dir.z);
+	float qx = 0;
+	float qy = 1 * sin(angle / 2);
+	float qz = 0;
+	float qw = cos(angle / 2);
+	return Quat(qx, qy, qz, qw);
+}
+
+void UltimateFang::OnCollision(GameObject& collidedWith, float3 collisionNormal, float3 penetrationDistance, void* particle) {
+	if (tickOn) {
+		if (collidedWith.name == "MeleeGruntWeaponized" || collidedWith.name == "RangedGrunt") {
+			collisionedGameObject.push_back(collidedWith);
+			Debug::Log(collidedWith.name.c_str());
+		}
+	}
+}
+
+void UltimateFang::StartUltiamte()
+{
+	ComponentSphereCollider* sphereCollider = GetOwner().GetComponent<ComponentSphereCollider>();
+	sphereCollider->radius = radius;
+	//App->physics->UpdateSphereRigidbody(sphereCollider); //TODO We need this function to be exposed, then we can use it
```

Comment:
I think it is already exposed in the current version of develop?

Reply:
Noice